linera-witty-macros/src/unit_tests/wit_store.rs

```diff
@@ -129,3 +129,107 @@ fn tuple_struct() {
 
     assert_eq!(output.to_string(), expected.to_string());
 }
+
+/// Check the generated code for the body of the implementation of `WitStore` for a enum.
+#[test]
+fn enum_type() {
+    let input: ItemEnum = parse_quote! {
+        enum Enum {
+            Empty,
+            Tuple(i8, CustomType),
+            Struct {
+                first: (),
+                second: String,
+            },
+        }
+    };
+    let output = derive_for_enum(&input.ident, input.variants.iter());
+
+    let expected = quote! {
+        fn store<Instance>(
+            &self,
+            memory: &mut linera_witty::Memory<'_, Instance>,
+            mut location: linera_witty::GuestPointer,
+        ) -> Result<(), linera_witty::RuntimeError>
+        where
+            Instance: linera_witty::InstanceWithMemory,
+            <Instance::Runtime as linera_witty::Runtime>::Memory:
+                linera_witty::RuntimeMemory<Instance>,
+        {
+            match self {
+                Enum::Empty => {
+                    0_u8.store(memory, location)?;
+                    location = location
+                        .after::<u8>()
+                        .after_padding_for::<linera_witty::HList![]>()
```

Comment:
Unrelated but could we rename `after_padding_for` into `after` and only use that one?

Reply:
I'm not sure I understood, but I think that wouldn't be a good idea. `after::<X>()` advances the pointer `X::SIZE`, while `after_padding_for::<Y>()` advances the pointer to align to `Y::Layout::ALIGNMENT`. `X` and `Y` are usually different types, and therefore we would need to make `after::<X, Y>()`, which seems more confusing to me.

Reply:
nevermind I was confused